RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc

```diff
@@ -354,27 +351,26 @@ void ProtonReconstructionAlgorithm::reconstructFromSingleRP(
     if (verbosity_)
       os << "\n    xi = " << xi << " +- " << xi_unc << ", th_y = " << th_y << " +- " << th_y_unc << ".";
 
-    using EX = reco::ProtonTrackExtra;
     using PT = reco::ProtonTrack;
 
     // save proton candidate
-    const bool valid = true;
-    reco::ProtonTrackExtra ptExtra(valid, EX::ReconstructionMethod::singleRP, { track });
 
     const double sign_z = (CTPPSDetId(track->getRPId()).arm() == 0) ? +1. : -1.;  // CMS convention
-    const reco::Track::Point vertex(0., 0., 0.);
+    const reco::ProtonTrack::Point vertex(0., 0., 0.);
     const double cos_th = sqrt(1. - th_y*th_y);
     const double p = lhcInfo.energy() * (1. - xi);
-    const reco::Track::Vector momentum(0., p * th_y, sign_z * p * cos_th);
+    const reco::ProtonTrack::Vector momentum(0., p * th_y, sign_z * p * cos_th);
 
-    reco::TrackBase::CovarianceMatrix cm;
+    reco::ProtonTrack::CovarianceMatrix cm;
     cm((int)PT::Index::xi, (int)PT::Index::xi) = xi_unc * xi_unc;
     cm((int)PT::Index::th_y, (int)PT::Index::th_y) = th_y_unc * th_y_unc;
 
     reco::ProtonTrack pt(0., 0, vertex, momentum, xi, cm);
```

Comment:
TODO: move it to the `reco::ProtonTrack` constructor?

Reply:
Woops, wrong line (3 below)

Reply:
I am confused - move what?

Reply:
Yes, this makes sense.